src/enclave/Enclave/ExpressionEvaluation.h

```diff
@@ -702,6 +702,59 @@ class FlatbuffersExpressionEvaluator {
       }
     }
 
+    case tuix::ExprUnion_In:
+    {
+      auto c = static_cast<const tuix::In *>(expr->expr());
+      size_t num_children = c->children()->size(); 
+      bool result = false;
+      if (num_children < 2){
+        throw std::runtime_error(std::string("In can't operate with an empty list, currently we have ")
+          + std::to_string(num_children - 1)
+          + std::string("items in the list")); 
+      }
+
+      auto left_offset =  eval_helper(row, (*c->children())[0]);
+      const tuix::Field *left = flatbuffers::GetTemporaryPointer(builder, left_offset);
+
+      bool result_is_null = left->is_null(); 
+
+      for (size_t i=1; i<num_children; i++){
+        auto right_offset = eval_helper(row, (*c->children())[i]);
+        const tuix::Field *item = flatbuffers::GetTemporaryPointer(builder, right_offset);
+        if (item->value_type() != left->value_type()){
+          throw std::runtime_error(
+          std::string("In can't operate on ")
+          + std::string(tuix::EnumNameFieldUnion(left->value_type()))
+          + std::string(" and ")
+          + std::string(tuix::EnumNameFieldUnion(item->value_type()))
+          + ". Please double check the type of each input");
+        }
+        result_is_null = result_is_null || item ->is_null(); 
```

Comment:
Will this set the result to be NULL if any item is NULL? Even if there is a match after the NULL value is iterated over?

Reply:
fixed. Now it's only null when there's no match and there are null inputs